shell/agents/AIShell.OpenAI.Agent/Service.cs

```diff
@@ -116,38 +118,54 @@ private void RefreshOpenAIClient()
             && string.Equals(old.Endpoint, _gptToUse.Endpoint)
             && string.Equals(old.Deployment, _gptToUse.Deployment)
             && string.Equals(old.ModelName, _gptToUse.ModelName)
-            && old.Key.IsEqualTo(_gptToUse.Key))
+            && old.AuthType == _gptToUse.AuthType
+            && (old.AuthType == AuthType.EntraID || old.Key.IsEqualTo(_gptToUse.Key)))
         {
-            // It's the same same endpoint, so we reuse the existing client.
+            // It's the same endpoint and auth type, so we reuse the existing client.
             return;
         }
 
         EndpointType type = _gptToUse.Type;
-        string userKey = Utils.ConvertFromSecureString(_gptToUse.Key);
 
         if (type is EndpointType.AzureOpenAI)
         {
             // Create a client that targets Azure OpenAI service or Azure API Management service.
             var clientOptions = new AzureOpenAIClientOptions() { RetryPolicy = new ChatRetryPolicy() };
             bool isApimEndpoint = _gptToUse.Endpoint.EndsWith(Utils.ApimGatewayDomain);
 
-            if (isApimEndpoint)
+            if (_gptToUse.AuthType == AuthType.ApiKey)
             {
-                clientOptions.AddPolicy(
-                    ApiKeyAuthenticationPolicy.CreateHeaderApiKeyPolicy(
-                        new ApiKeyCredential(userKey),
-                        Utils.ApimAuthorizationHeader),
-                    PipelinePosition.PerTry);
-            }
+                string userKey = Utils.ConvertFromSecureString(_gptToUse.Key);
 
-            string azOpenAIApiKey = isApimEndpoint ? "placeholder-api-key" : userKey;
+                if (isApimEndpoint)
+                {
+                    clientOptions.AddPolicy(
+                        ApiKeyAuthenticationPolicy.CreateHeaderApiKeyPolicy(
+                            new ApiKeyCredential(userKey),
+                            Utils.ApimAuthorizationHeader),
+                        PipelinePosition.PerTry);
+                }
 
-            var aiClient = new AzureOpenAIClient(
-                new Uri(_gptToUse.Endpoint),
-                new ApiKeyCredential(azOpenAIApiKey),
-                clientOptions);
+                string azOpenAIApiKey = isApimEndpoint ? "placeholder-api-key" : userKey;
 
-            _client = aiClient.GetChatClient(_gptToUse.Deployment);
+                var aiClient = new AzureOpenAIClient(
+                    new Uri(_gptToUse.Endpoint),
+                    new ApiKeyCredential(azOpenAIApiKey),
+                    clientOptions);
+
+                _client = aiClient.GetChatClient(_gptToUse.Deployment);
+            }
+            else if (_gptToUse.AuthType == AuthType.EntraID)
```

Comment:
No need to compare here.
```suggestion
            else
```